src/main/java/org/testobject/appium/junit/TestObjectTestResultWatcher.java

```diff
@@ -1,186 +1,61 @@
 package org.testobject.appium.junit;
 
-import io.appium.java_client.AppiumDriver;
-import jersey.repackaged.com.google.common.base.Optional;
 import org.junit.internal.AssumptionViolatedException;
 import org.junit.rules.TestWatcher;
 import org.junit.runner.Description;
-import org.openqa.selenium.OutputType;
 import org.openqa.selenium.remote.RemoteWebDriver;
-import org.testobject.appium.junit.internal.Util;
-import org.testobject.rest.api.RestClient;
+import org.testobject.appium.IntermediateReporter;
 import org.testobject.rest.api.appium.common.TestObjectCapabilities;
-import org.testobject.rest.api.appium.common.data.SuiteReport;
-import org.testobject.rest.api.appium.common.data.Test;
-import org.testobject.rest.api.appium.common.data.TestReport;
-import org.testobject.rest.api.appium.common.data.TestResult;
-import org.testobject.rest.api.resource.AppiumResource;
-import org.testobject.rest.api.resource.AppiumSuiteReportResource;
 
 import java.net.MalformedURLException;
 import java.net.URL;
 
-import static org.testobject.rest.api.appium.common.TestObjectCapabilities.TESTOBJECT_API_ENDPOINT;
-import static org.testobject.rest.api.appium.common.TestObjectCapabilities.TESTOBJECT_API_KEY;
-
 public class TestObjectTestResultWatcher extends TestWatcher {
 
-	private final String baseUrl;
-
-	private RemoteWebDriver remoteWebDriver;
-	private URL driverRemoteAddress;
-
-	private RestClient client;
-
-	private String apiKey;
-	private long suiteId;
-	private SuiteReport suiteReport;
-	private Test test;
+	private IntermediateReporter reporter;
 
 	public TestObjectTestResultWatcher() {
-		this(TESTOBJECT_API_ENDPOINT);
+		this(false);
 	}
 
-	public TestObjectTestResultWatcher(String baseUrl) {
-		this.baseUrl = baseUrl;
+	public TestObjectTestResultWatcher(boolean isTestLocal) {
```

Comment:
Do folks no longer have the ability to set a custom URL? What if their local Appium server is on a non-standard port?

You could consider having an overloaded constructor that takes a URL or string, or maybe a factory with methods like .create(), .createLocal(), create(String)

edit: nevermind the other constructors are below